filterExprParser.go

```diff
@@ -102,14 +102,9 @@ func (fe *FilterExpression) String() string {
 	return strings.Join(output, " ")
 }
 
-// Outputs the head of the Expression match tree of which represents everything underneath
-func (f *FilterExpression) OutputExpression() (Expression, error) {
+func (f *FilterExpression) outputExpressionNoParenCheck() (Expression, error) {
```

Comment:
skipping check at subexpression may not be safe.

the following expression is "OK" at top level since it has the same number of open and close parenthesis, but it is not valid. such invalidness can only be detected at subexpression check level.

( (subexpr( )subexpr2) )

if the expr above is not possible with the way we parse expressions, then it may be ok. not sure about this , though

Reply:
Good news is that the parser grammar should enforce any invalid parenthesis use cases...
`// AndCondition             = { OpenParens } Condition { "AND" Condition } { CloseParen }`
I've verified that the neg case you mentioned is validated correctly.